flytekit/extend/backend/base_agent.py

```diff
@@ -198,34 +218,30 @@ async def _create(
         self, task_template: TaskTemplate, output_prefix: str, inputs: typing.Dict[str, typing.Any] = None
     ) -> CreateTaskResponse:
         ctx = FlyteContext.current_context()
-        grpc_ctx = _get_grpc_context()
 
         # Convert python inputs to literals
         literals = inputs or {}
         for k, v in inputs.items():
             literals[k] = TypeEngine.to_literal(ctx, v, type(v), self._entity.interface.inputs[k].type)
-
-        literal_map = LiteralMap(literals)
-
-        if isinstance(self, PythonFunctionTask):
+        literal_map = LiteralMap(literals) if literals else None
+        if literal_map and isinstance(self, PythonFunctionTask):
```

Comment:
`pyflyte-execute` has some issues, so we should always upload `literals.pb` even if the literal is None.
check out https://github.com/flyteorg/flytekit/pull/2054

Reply:
Thank you!